### DISCUSSION TITLE
Custom Window for dialogs

### DISCUSSION THREAD

**Discussion by cirrusone** [AvaloniaCommunity/Prism.Avalonia#94] [Q&A] [answered]
Great project, thank you.

Are there any samples showing how to use  a custom Window to host a dialog rather than the inbuilt `Prism.Avalonia/Services/Dialogs/DialogWindow.axaml`?

I can see IDialogService allows passing name of the hosting window registered but I haven't been able to get this working using

```cs
    protected override void RegisterTypes(IContainerRegistry containerRegistry)
    {
        containerRegistry.Register<MainWindow>();
        containerRegistry.Register<CustomMainWindow>(); // Not working CustomMainWindow.axaml?
        containerRegistry.RegisterDialog<MessageBoxView, MessageBoxViewModel>();
        containerRegistry.RegisterDialog<DialogView, DialogViewModel>();
    }
```

It would be useful to be able to style the parent window inline with the main application (eg ExtendClientAreaToDecorationsHint="True"). I can see there is a similar open issue relating to style  [Add dialog style Settings #84](https://github.com/AvaloniaCommunity/Prism.Avalonia/issues/84) but custom window would  be more useful in certain situations

**Comment by ByteSquire** [comment, marked as answer]
You need to register a DialogWindow using the [RegisterDialogWindow](https://github.com/AvaloniaCommunity/Prism.Avalonia/blob/c3524071eb97202b7ba885783f00a34c8b914ca1/src/Prism.Avalonia/Ioc/IContainerRegistryExtensions.cs#L50C39-L50C39) method like [this](https://github.com/ByteSquire/TQDB-Editor/blob/fa354366628d7d282bb2518481ba4ec440f65b89/TQDBEditor/Dialogs/Module.cs#L21):
`containerRegistry.RegisterDialogWindow<CustomMainWindow>();`
, or provide a name if you don't want to override the default dialog window.
Make sure your window implements [IDialogWindow](https://github.com/AvaloniaCommunity/Prism.Avalonia/blob/c3524071eb97202b7ba885783f00a34c8b914ca1/src/Prism.Avalonia/Services/Dialogs/IDialogWindow.cs#L12) like [this](https://github.com/ByteSquire/TQDB-Editor/blob/fa354366628d7d282bb2518481ba4ec440f65b89/TQDBEditor/Dialogs/BaseDialogWindow.axaml.cs#L6):
`class CustomMainWindow : Window, IDialogWindow`
Btw, take a look at [this Style definition](https://github.com/ByteSquire/TQDB-Editor/blob/fa354366628d7d282bb2518481ba4ec440f65b89/TQDBEditor/App.axaml#L31C1-L40) if you want to have a consistent look for your application's windows.

**Comment by cirrusone** [comment]
Excellent, thank you. I was close but needed to add `IDialogWindow` and the corresponding interface method `IDialogResult?` to the custom window.
